src/native/libs/System.Native/pal_signal.c

```diff
@@ -313,12 +314,12 @@ static void* SignalHandlerLoop(void* arg)
     free(arg);
     assert(pipeFd >= 0);
 
-    char* threadName = ".NET SigHandler";
-#if defined(__linux__) || defined(__FreeBSD__)
-    pthread_setname_np(pthread_self(), threadName);
-#endif
-#if defined(__APPLE__)
-    pthread_setname_np(threadName);
+#ifndef TARGET_WASM
+    // WASM does not support pthread_setname_np yet: https://github.com/emscripten-core/emscripten/pull/18751
```

Comment:
Mono has a typescript based implementation in `src/mono/mono/utils/mono-threads-wasm.c` for `mono_native_thread_set_name`, which is limited. I didn't wanted to mix these because mono has different TARGET_ macros (those are HOST_). this is only covering posix for now.